kernel/src/common/deferred_call.rs

```diff
@@ -4,18 +4,51 @@
 //! in the chip scheduler if the hardware doesn't support interrupts where
 //! they are needed.
 
+use core::cell::UnsafeCell;
 use core::convert::Into;
 use core::convert::TryFrom;
 use core::convert::TryInto;
+use core::intrinsics;
 use core::marker::Copy;
-use core::sync::atomic::AtomicUsize;
-use core::sync::atomic::Ordering;
+use core::marker::Sync;
+
+/// AtomicUsize with no CAS operations that works on targets that have "no atomic
+/// support" according to their specification. This makes it work on thumbv6
+/// platforms.
+///
+/// Borrowed from https://github.com/japaric/heapless/blob/master/src/ring_buffer/mod.rs
+/// See: https://github.com/japaric/heapless/commit/37c8b5b63780ed8811173dc1ec8859cd99efa9ad
+struct AtomicUsize {
```

Comment:
could maybe make sense to extract `AtomicUsizeWithoutCAS` to a separate module, might be needed by other code